### PR TITLE
implemented comment section on article

### PR DESCRIPTION
**Pull request by olivershh**

### CODE REVIEW

**Review by xmenbteam — APPROVED** [review]
This is coming along really well! Well done :)

**Comment by olivershh** [comment]
Kind of a mess here as I clearly rushed and I think didnt confirm the previous merge? Just trying to wrap my head around it.

**Comment by olivershh** [comment]
Accidentily broke it but fixed again with the above push

File: src/components/CommentList.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const [isLoading, setIsLoading] = useState(true)
+
+
+useEffect(() => {
+
+    getCommentsByArticleID(article_id).then(({comments}) => {
+        setIsLoading(false)
+        setComments(comments)
+    })
```
Comment (xmenbteam) [inline src/components/CommentList.jsx:17]: `.catch` please!

File: src/components/PostComment.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    const handleSubmit = (event) => {
+        event.preventDefault()
+        if (userComment === "") return
+        setComments(prevComments => {
+            return [{author: "grumpy19", body: userComment, votes: 0}, ...prevComments]
+        })
+        setUserComment("")
+        postCommentByID(article_id, userComment)
```
Comment (xmenbteam) [inline src/components/PostComment.jsx:20]: This needs a `.then` and a `.catch` after it please!

File: src/components/api-calls.js [hunk 3]
```diff
@@ -33,3 +33,16 @@ export function increaseVotes(article_id) {
...
 }
+
+export function getCommentsByArticleID(article_id) {
+  return api.get(`/articles/${article_id}/comments`).then(({ data }) => {
+    return data;
+  });
+}
+
+export function postCommentByID(article_id, comment) {
+  const postObject = { usernme: "grumpy19", body: comment };
```
Comment (xmenbteam) [inline src/components/api-calls.js:44]: You spelled username wrong. You'd see this if you had a `.catch` in your component!

**Review by xmenbteam — CHANGES_REQUESTED** [review]
Because your comments are paginated in the back end, they're coming back in the wrong order so you're not seeing the most recent ones. Have a go at changing either the default order in the backend or changing the order they're queried from your front end?

File: src/components/CommentList.jsx (hunk 1 above)
Comment (olivershh) [reply src/components/CommentList.jsx:17]: Sorted! Not dealing with the error yet, but think that will be covered in a later ticket

File: src/components/PostComment.jsx (hunk 2 above)
Comment (olivershh) [reply src/components/PostComment.jsx:19]: Sorted

File: src/components/api-calls.js (hunk 3 above)
Comment (olivershh) [reply src/components/api-calls.js]: Sorted thanks.

**Comment by olivershh** [comment]
Sorted backend to display by created_at descending
Also added a temp comment_id to new comments (used Date.now() since thats the key Im giving react.

**Pull request merged**